source/GlassView.Core.Test/Serialization/SerializationTest.cs

```diff
@@ -0,0 +1,63 @@
+using Atmoos.GlassView.Core.Models;
+
+namespace Atmoos.GlassView.Core.Test;
+
+public class SerializationTest
+{
+    private static readonly String testFile = Path.Combine("Resources", "TestBenchmark.json");
+
+    [Fact]
+    public void SerializationIsPossible()
+    {
+        String expected = File.ReadAllText(testFile);
+        BenchmarkSummary benchmarks = expected.Deserialize<BenchmarkSummary>();
+
+        String actual = benchmarks.Serialize();
+
+        Assert.Equal(expected, actual, ignoreAllWhiteSpace: true, ignoreLineEndingDifferences: true);
+    }
+
+    [Fact]
+    public void DeserializationIsPossible()
+    {
+        BenchmarkSummary actual = File.ReadAllText(testFile).Deserialize<BenchmarkSummary>();
+
+        Assert.NotNull(actual);
+        Assert.Equal(actual.Count, actual.Count);
+    }
+
+    [Fact]
+    public void RoundRobinSerializationSucceeds()
+    {
+        BenchmarkSummary expected = File.ReadAllText(testFile).Deserialize<BenchmarkSummary>();
+
+        String serializedSummary = expected.Serialize();
+        BenchmarkSummary actual = serializedSummary.Deserialize<BenchmarkSummary>();
+
+        AssertEqual(expected, actual);
+    }
+
+    // ToDo: For some reason the standard equality check fails, so asserting each property individually for now...
+    private static void AssertEqual(BenchmarkSummary expected, BenchmarkSummary actual)
```

Comment:
I think the standard equality fails because the BenchmarkSummary contains arrays which are always compared by reference.
If we implement [IEquatable ](https://learn.microsoft.com/en-us/dotnet/api/system.iequatable-1?view=net-8.0) on BenchmarkSummary, it should work.

Reply:
Naturally you were right regarding the arrays. They are the root cause for Equality of record types not to follow "true" value  semantics.
(I would have expected the compiler to solve this. Hence, my initial comment)

Reply:
I'm fine not implementing `IEquatable` on our record classes. We currently would only use that in the tests. And it's error prone to implement reliably ourselves, as anytime a property is added, we'd need to manually scan the impl. of our records or have a randomised test suite that probes all possible combinations of properties to reliably catch which properties we're missing in our impl.
To me more of a maintenance issue than something useful.

Is it ok for you if we don't implement that interface?

Reply:
Since we use this only in tests, and the actual users will not need to compare BenchmarkSummaries, I'm fine with not implementing `IEquatable`.